livre/lib/cache/HistogramObject.cpp

```diff
@@ -40,14 +41,48 @@ void binData( const SRC_TYPE* rawData,
               const size_t compCount,
               const uint64_t scaleFactor )
 {
-    const Vector2f dataSourceRange( std::numeric_limits< SRC_TYPE >::min(),
-                                    std::numeric_limits< SRC_TYPE >::max());
-    const double binCount = histogram.getBinsSize();
-    const double minVal = dataSourceRange[ 0 ];
-    const double maxVal = dataSourceRange[ 1 ];
-    const double range = maxVal - minVal;
-
-    uint64_t* dstData = histogram.getBins();
+    const double binCount = histogram.getBins().size();
+    uint64_t* dstData = histogram.getBins().data();
+
+    if( std::is_integral< SRC_TYPE >::value )
+    {
+        histogram.setMin( std::numeric_limits< SRC_TYPE >::min( ));
+        histogram.setMax( std::numeric_limits< SRC_TYPE >::max( ));
+    }
+    else
+    {
+        double minVal = histogram.getMin();
+        double maxVal = histogram.getMax();
+
+        const Vector3ui dataBlockSize = blockSize + padding * 2;
+        for( size_t i = padding.x(); i < dataBlockSize.x() - padding.x(); ++i )
+            for( size_t j = padding.y(); j < dataBlockSize.y() - padding.y(); ++j )
+                for( size_t k = padding.z(); k < dataBlockSize.z() - padding.z(); ++k )
+                    for( size_t c = 0; c < compCount; ++c )
+                    {
+                        const size_t index = compCount * i * dataBlockSize.y() * dataBlockSize.z() +
+                                             compCount * j * dataBlockSize.z() +
+                                             compCount * k + c;
+                        const double data = rawData[ index ];
+                        if( data < minVal )
+                            minVal = data;
+                        if( data > maxVal )
+                            maxVal = data;
+                    }
+
+        histogram.setMin( minVal );
+        histogram.setMax( maxVal );
+
+        if(( maxVal - minVal ) == 0.0f )
+        {
+            histogram.getBins().clear();
+            const size_t bins = ( dataBlockSize - padding ).product() * scaleFactor * compCount;
```

Comment:
What is this scaleFactor? In line 98 below you add, it but here you are multiplying.